tests/lib/files.sh

```diff
@@ -55,6 +55,10 @@ ensure_file_exists_backup_real() {
     if [ -f "$file" ]; then
         mv "$file" "$file.back"
     fi
+    # ensure the parent dir is available
+    if [ ! -d "$(dirname $file)" ]; then
+        mkdir -p "$(dirname $file)"
```

Comment:
We could just call `mkdir -p ...` since that's exactly what it means anyway. Not a blocker, just calling it out.